source/Octopus.Tentacle.Kubernetes.Tests.Integration/KubernetesAgent/KubernetesAgentMigrateFromPreinstallationTest.cs

```diff
@@ -0,0 +1,189 @@
+using System.Text;
+using FluentAssertions;
+using k8s;
+using k8s.Models;
+using Octopus.Diagnostics;
+using Octopus.Tentacle.Commands;
+using Octopus.Tentacle.Diagnostics;
+using Octopus.Tentacle.Startup;
+
+namespace Octopus.Tentacle.Kubernetes.Tests.Integration.KubernetesAgent;
+
+public class KubernetesAgentMigrateFromPreinstallationTest
```

Comment:
In real cases - is there a scenario where there are no destination resources and we end up creating a new secret and configmap?
If so, I think it would be good to test that case too.

Reply:
No - the configmaps and secrets are made by the helm chart, and never tentacle